File: frontend/src/services/warehouseService.ts
import { apiClient } from '@/config/apiClient';
import type { 
  ApiResponse, 
  PageResponse, 
  WarehouseResponse, 
  WarehouseRequest, 
  LocationNodeResponse,
  TableParams 
} from '@/types';

export const warehouseService = {
  getAll: (params?: TableParams): Promise<ApiResponse<PageResponse<WarehouseResponse>>> => {
    return apiClient.get('/warehouses', { params });
  },

  getById: (id: number): Promise<ApiResponse<WarehouseResponse>> => {
    return apiClient.get(`/warehouses/${id}`);
  },

  create: (data: WarehouseRequest): Promise<ApiResponse<WarehouseResponse>> => {
    return apiClient.post('/warehouses', data);
  },

  update: (id: number, data: WarehouseRequest): Promise<ApiResponse<WarehouseResponse>> => {
    return apiClient.put(`/warehouses/${id}`, data);
  },

  delete: (id: number): Promise<ApiResponse<void>> => {
    return apiClient.delete(`/warehouses/${id}`);
  },

  getLocationTree: (id: number): Promise<ApiResponse<LocationNodeResponse[]>> => {
    return apiClient.get(`/warehouses/${id}/locations/tree`);
  },
};

File: backend/src/main/java/com/scim/service/WarehouseService.java
package com.scim.service;

import com.scim.dto.request.WarehouseRequest;
import com.scim.dto.response.PageResponse;
import com.scim.dto.response.WarehouseResponse;
import org.springframework.data.domain.Pageable;

public interface WarehouseService {

    PageResponse<WarehouseResponse> getAll(String search, Pageable pageable);

    WarehouseResponse getById(Long id);

    WarehouseResponse create(WarehouseRequest request);

    WarehouseResponse update(Long id, WarehouseRequest request);

    void delete(Long id);
}
